content/Dev/remote_settings_history.rst

```diff
@@ -0,0 +1,179 @@
+The History of Firefox Remote Settings
+######################################
+
+:tags: python, javascript, rust
+:date: 2020-10-19
+
+
+When I started to write the first lines of this article, it was the last day of Ethan in our team. His departure marked the end of an era that I'll try to tell you about here.
+
+
+Prehistory (~2014)
+==================
+
+While I was working at `Makina Corpus <https://makina-corpus.com>`_, we were regularly coming up with the same kind of architectures: a database, a REST API, and frontend/mobile stuff. Other teams were also starting to consider the idea of «headless CMS».
+
+I wanted to build a reusable backend, where you define your models in JSON and retrieve/store records.
+Alexis and I wrote down some dream code at a DjangoCon and we started the `Daybed project <https://github.com/spiral-project/daybed/>`_, using Pyramid. It became functional and I even talked about it at `FOSDEM <https://archive.fosdem.org/2015/schedule/event/daybed/>`_.
+
+Alexis, Rémy, and Tarek were building Web APIs in the Cloud Services team, and there were needs to offer remote storage features to `Firefox OS <https://en.wikipedia.org/wiki/Firefox_OS>`_ app developers.
+
+I was hired in their team and we were about to roll out our first app: a «reading list» service (à-la Pocket).
+
+
+Neolithic (2015-2016)
+=====================
+
+The team had accumulated a lot of experience with Web APIs, and one thing was certain: we shouldn't start each project from scratch. Even though we had a secret plan (a reusable service), we couldn't justify starting with that.
+
+We gathered a lot of good practices and a handful tooling in one library: `Cliquet <https://cliquet.readthedocs.io/en/latest/>`_. We hated the word «framework», but at least we wanted a standard way of building services (cf. `talk at PyBCN <https://mozilla-services.github.io/cliquet/talks/2015.07.pybcn/>`_).
+
+.. image:: images/remote-settings-history-cliquet.png
+
+*Cliquet* had the notion of pluggable storage backends. And, because we had this vision of reusable services, we built a proof-of-concept where a service built with *Cliquet* would be storing its data in another service built with *Cliquet* 🤯 [#]_.
+
+Eventually, this storage service became `Kinto <https://github.com/Kinto/>`_, and *Cliquet* became ``kinto.core``.
+
+`n1k0 <https://nicolas.perriault.net/>`_ and `Magopian <https://mathieu.agopian.info/>`_ joined the team, and built a nice offline-first SDK for Kinto. Plus a generic UI to CRUD records of any accessible Kinto server (kind of remote phpmyadmin). In order to generate forms from collections JSON schema, they released `react-jsonschema-form <https://github.com/rjsf-team/react-jsonschema-form/>`_ (which now has like 9k stars on Github!).
+
+This was an exciting time! Kinto was on Hackernews [#]_ [#]_, and the number of Github stars was going up!
```

Comment:
I'm having a hard time figuring out what `[#]_ [#]_` does, despite being a funny emoji. Just wanted to check it's not a mistake.